src/elementPlugins/shortText/src/ApplicationView.tsx

```diff
@@ -9,43 +9,70 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
   isEditable,
   isRequired,
   allResponses,
+  evaluator,
 }) => {
-  const [validationMessage, setValidationMessage] = useState('')
+  const [validationMessageDisplay, setValidationMessageDisplay] = useState('')
   const [value, setValue] = useState(initialValue?.text)
+  const [isValid, setIsValid] = useState<boolean>()
+  const [responses, setResponses] = useState({
+    thisResponse: undefined,
+    ...allResponses,
+  })
+  const {
+    validation: validationExpression,
+    validationMessage,
+    placeholder,
+    maskedInput,
+  } = templateElement?.parameters
 
-  return (
-    <Form.Input
-      fluid
-      label={templateElement.title}
-      placeholder={templateElement.parameters.placeholder}
-      onChange={onChange(setValue, setValidationMessage, onUpdate)}
-      value={value}
-      disabled={!isEditable}
-      error={
-        validationMessage
-          ? {
-              content: validationMessage,
-              pointing: 'above',
-            }
-          : null
-      }
-    />
-  )
-}
+  useEffect(() => {
+    // Do validation, setIsValid
+    if (validationExpression && responses.thisResponse !== undefined) {
+      evaluator(validationExpression, { objects: [responses] }).then((result: boolean) => {
+        setIsValid(result)
+      })
+    } else setIsValid(true)
+  }, [responses])
+
+  useEffect(() => {
+    setResponses({ thisResponse: value, ...allResponses })
+  }, [value])
+
+  useEffect(() => {
+    if (isValid) setValidationMessageDisplay('')
+    else setValidationMessageDisplay(validationMessage)
+  }, [isValid])
 
-function onChange(updateValue: any, updateValidationMessage: any, onUpdate: any) {
-  return (_: any, { value }: any) => {
-    updateValue(value)
-    // Validation here is just an example.
-    // ideally we use validation condition from templateElement and query evaluator
-    if (value.match(/^[a-zA-Z ]*$/) == null) {
-      updateValidationMessage('Should only have letters and space')
-      onUpdate({ isValid: false })
-    } else {
-      updateValidationMessage('')
-      onUpdate({ value: value, isValid: true })
-    }
+  function handleChange(e: any) {
+    setValue(e.target.value)
   }
+
+  function handleLoseFocus(e: any) {
+    onUpdate({ value: { text: e.target.value }, isValid })
```

Comment:
shouldn't this just be ` {value, isValid} ` ?

Reply:
No, because we need to send the updated Value in the correct Response object back to onUpdate, which is:
```
{
  text: <actual value string>
  option?: int
  ...others?:any
}
```

Previously I was just sending the raw value up and constructing this object in the onUpdate call, but it's better to do it here since the plugin itself will need to provide this additional information. It doesn't make any difference in this plugin, but for selectors, etc., this will be necessary.

Reply:
cool, I guess I was thinking to still use the value from state (but still send it back in object)

Reply:
Oh, right, you mean just:
```
 function handleLoseFocus(e: any) {
    onUpdate({ value: { text: value }, isValid })
```

Done 7d2261c9